### PR TITLE
Improve docs

### PR DESCRIPTION
**Pull request by chaws**
Been a while since i've been wanting to add a note on these. Close https://github.com/Linaro/squad/issues/459, https://github.com/Linaro/squad/issues/461 and https://github.com/Linaro/squad/issues/460

### CODE REVIEW

File: doc/api.rst [hunk 1]
```diff
@@ -287,12 +287,23 @@ following additional routes:
...
 
 tests (/api/tests/)
 ~~~~~~~~~~~~~~~~~~~
 
-Provides access to Test object. In case of private projects token with
-enough privileges is required to access the object.
+Provides access to Tests objects. In case of private projects token with
+enough privileges is required to access the objects.
+
+metrics (/api/tests/)
```
Comment (mwasilew) [inline doc/api.rst:302]: /api/metrics

File: doc/intro.rst [hunk 2]
```diff
@@ -40,6 +40,13 @@ will send notification to project subscribers if the test result metrics exceed
 a certain value. The threshold values will also appear in the charts. Projects 
 have the `project_settings` field for any specific configuration it might require.
 
+Builds can be compared against others, exposing regressions and fixes. Visit
```
Comment (mwasilew) [inline doc/intro.rst:43]: I propose this wording: "Builds can be compared against baselines exposing regressions and fixes."

File: doc/intro.rst [hunk 3]
```diff
@@ -40,6 +40,13 @@ will send notification to project subscribers if the test result metrics exceed
...
+``/_/comparebuilds/``, select a project then two builds from it. This comparison
+will go over each build's tests and find all different states from each. For instance
+if a test failed for an older build but passes in a more current one, this is considered
```
Comment (mwasilew) [inline doc/intro.rst:46]: a test failed in baseline but passes in a current build...

File: doc/api.rst (hunk 1 above)
Comment (chaws) [reply doc/api.rst:302]: done

File: doc/intro.rst (hunk 2 above)
Comment (chaws) [reply doc/intro.rst:43]: Agree, "baseline" is more abstract

File: doc/intro.rst (hunk 3 above)
Comment (chaws) [reply doc/intro.rst:46]: Done

**Pull request merged**